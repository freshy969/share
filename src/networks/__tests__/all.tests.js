import { types, unprotect } from 'mobx-state-tree';
import { set } from 'mobx';
import All from '../all';
import Network from '../network';

describe('Share > All', () => {
  test('count', () => {
    const share = types
      .model('Share')
      .props({
        all: types.optional(All, {}),
        network1: types.optional(Network, {}),
        network2: types.optional(Network, {}),
      })
      .views(self => ({
        get networks() {
          return [self.network1, self.network2];
        },
      }))
      .create({});

    unprotect(share);

    expect(share.all.count({ type: 'post', id: 60 })).toBe(null);

    set(share.network1.countsMap, { post_60: 123 });
    expect(share.network1.count({ type: 'post', id: 60 })).toBe(123);
    expect(share.all.count({ type: 'post', id: 60 })).toBe(123);

    set(share.network2.countsMap, { post_60: 111 });
    expect(share.network2.count({ type: 'post', id: 60 })).toBe(111);
    expect(share.all.count({ type: 'post', id: 60 })).toBe(234);
  });

  test('requestCount', async () => {
    const requestCount = jest.fn();
    const NetworkMock = Network.actions(() => ({ requestCount }));

    const share = types
      .model('Share')
      .props({
        all: types.optional(All, {}),
        network1: types.optional(NetworkMock, {}),
        network2: types.optional(NetworkMock, {}),
      })
      .views(self => ({
        get networks() {
          return [self.network1, self.network2];
        },
      }))
      .create({});

    share.all.requestCount({ type: 'post', id: 60 });
    expect(requestCount.mock.calls.length).toBe(share.networks.length);
    expect(requestCount.mock.calls).toMatchSnapshot();
  });
});
