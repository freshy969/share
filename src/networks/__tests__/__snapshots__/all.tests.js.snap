// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share > All requestCount 1`] = `
Array [
  Array [
    Object {
      "id": 60,
      "type": "post",
    },
  ],
  Array [
    Object {
      "id": 60,
      "type": "post",
    },
  ],
]
`;
